[CI] Enable coverage report with Codecov

yunchu opened the pull request:
Replaced coverage report from Codacy to Codecov

Note that some tox usage is updated for the "pre-merge" env.
to run testing on your local machine, you need to set the testing target folder to the tox command like below.
(the default value is "tests/unit tests/integration")
```
tox -e pre-merge -- <testing-target-path>
```

yunchu committed:
Replace coverage reporting tool to Codecov

yunchu committed:
updated workflows

* added default positional args to pre-merge jobs
* added daily regression tests
* added on.push event to trigger pre-merge testing

samet-akcay commented:
@yunchu, great progress!

I have two questions:
- When I check https://app.codecov.io/gh/openvinotoolkit, I cannot see `training_extensions` there. Is it because we need to set this up first, or the repo has not yet been approved?
- With this codecov configuration, will it be possible to work on forked repos and still upload the coverage for PRs?
The reason for asking the second question is that ideally developers/PR makers should be working on forks rather than branches on the main repo. One potential roadblock, however, would be that coverage report may not be uploaded for PRs made via forks (a problem that anomalib has for example).

yunchu commented:
@samet-akcay 
actually it's not clear yet for me since I want to keep checking the status after merged it into the default branch. I'm an admin for this repo so I think I can add it to the codecov but I did not do it yet. please give little time to check all things are working as expected without having sign up to the external app.

yunchu committed:
updated badge section in readme

- removed codacy badges
- added codecov badge for coverage

yunchu committed:
updated ci workflow

- added weekly targetted to tests/regression
- added daily targetted to tests/e2e

yunchu committed:
update pre-merge jobs

- changed target path for integration testing to tests/integration/cli

goodsong81 reviewed (approved):
LGTM!

sungmanc reviewed (approved):
Thanks for the nice work
